SCons/Tool/ninja/NinjaState.py

```diff
@@ -0,0 +1,708 @@
+# MIT License
+#
+# Copyright The SCons Foundation
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY
+# KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import io
+import os
+import sys
+from os.path import splitext
+import ninja
+
+import SCons
+from SCons.Script import COMMAND_LINE_TARGETS
+from SCons.Util import is_List
+import SCons.Tool.ninja.Globals
+from .Globals import COMMAND_TYPES, NINJA_RULES, NINJA_POOLS, \
+    NINJA_CUSTOM_HANDLERS
+from .Rules import _install_action_function, _mkdir_action_function, _lib_symlink_action_function, _copy_action_function
+from .Util import get_path, alias_to_ninja_build, generate_depfile, ninja_noop, get_command, get_order_only, \
```

Comment:
since this is new code, may as well use Black-style setup for importing many symbols, then some future reformat won't cause a change:
```
from .Rules import (
    _install_action_function,
    _mkdir_action_function,
    _lib_symlink_action_function,
    _copy_action_function,
)
from .Util import (
    get_path,
    alias_to_ninja_build,
    generate_depfile,
    ninja_noop,
    get_command,
    get_order_only,
    get_outputs,
    get_inputs,
    get_dependencies,
    get_rule,
    get_command_env,
)
```